src/wallet/rpcwallet.cpp

```diff
@@ -2923,6 +2924,26 @@ UniValue listunspent(const JSONRPCRequest& request)
                 CScript redeemScript;
                 if (pwallet->GetCScript(hash, redeemScript)) {
                     entry.push_back(Pair("redeemScript", HexStr(redeemScript.begin(), redeemScript.end())));
+                    CTxDestination witness_destination;
+                    if(redeemScript.IsPayToWitnessScriptHash() && ExtractDestination(redeemScript, witness_destination)) {
+                        const WitnessV0ScriptHash& whash = boost::get<WitnessV0ScriptHash>(witness_destination);
+                        CScriptID id;
+                        CRIPEMD160().Write(whash.begin(), whash.size()).Finalize(id.begin());
+                        CScript witnessScript;
+                        if (pwallet->GetCScript(id, witnessScript)) {
+                            entry.push_back(Pair("witnessScript", HexStr(witnessScript.begin(), witnessScript.end())));
+                        }
+                    }
+                }
+            }
+
+            if (scriptPubKey.IsPayToWitnessScriptHash()) {
+                const WitnessV0ScriptHash& whash = boost::get<WitnessV0ScriptHash>(address);
+                CScriptID id;
+                CRIPEMD160().Write(whash.begin(), 32).Finalize(id.begin());
```

Comment:
nit: the `whash.size()` should also go here... :P

Reply:
Heh missed that, thanks :P